File: src/main/java/com/tiendayuliana/backend/model/Cliente.java
package com.tiendayuliana.backend.model;

import java.math.BigDecimal;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter @Setter @NoArgsConstructor @AllArgsConstructor
@Entity @Table(name = "cliente", schema = "tienda_yuliana")
public class Cliente {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_cliente")
    private Integer idCliente;

    public Integer getIdCliente() {
        return idCliente;
    }

    @Column(nullable = false, length = 120)
    private String nombre;

    @Column(length = 30)
    private String telefono;

    @Column(length = 30)
    private String nit;

    @Column(name = "es_mayorista", nullable = false)
    private boolean esMayorista = false;

    @Column(name = "limite_credito", nullable = false)
    private BigDecimal limiteCredito = BigDecimal.ZERO;

    @Column(name = "estado_credito", nullable = false, length = 15)
    private String estadoCredito = "ACTIVO";

    @Column(nullable = false)
    private Boolean activo = true;

    // Getter for esMayorista
    public boolean isEsMayorista() {
        return esMayorista;
    }

    // Setter for esMayorista
    public void setEsMayorista(boolean esMayorista) {
        this.esMayorista = esMayorista;
    }
}

File: frontend/lib/models/cliente.dart
class Cliente {
  final int idCliente;
  final String nombre;
  final String? telefono;
  final String? nit;
  final bool esMayorista;
  final double limiteCredito;
  final String estadoCredito; // ACTIVO | BLOQUEADO
  final bool activo;

  Cliente({
    required this.idCliente,
    required this.nombre,
    this.telefono,
    this.nit,
    required this.esMayorista,
    required this.limiteCredito,
    required this.estadoCredito,
    required this.activo,
  });

  factory Cliente.fromJson(Map<String, dynamic> json) => Cliente(
        idCliente: int.parse((json['idCliente'] ?? json['id'] ?? '0').toString()),
        nombre: (json['nombre'] ?? '').toString(),
        telefono: json['telefono']?.toString(),
        nit: json['nit']?.toString(),
        esMayorista: (json['esMayorista'] as bool?) ?? false,
        limiteCredito: (json['limiteCredito'] is num)
            ? (json['limiteCredito'] as num).toDouble()
            : double.tryParse(json['limiteCredito']?.toString() ?? '0') ?? 0,
        estadoCredito: (json['estadoCredito'] ?? 'ACTIVO').toString(),
        activo: (json['activo'] as bool?) ?? true,
      );

  Map<String, dynamic> toRequest() => {
        'nombre': nombre,
        if (telefono != null) 'telefono': telefono,
        if (nit != null) 'nit': nit,
        'esMayorista': esMayorista,
        'limiteCredito': limiteCredito,
        'estadoCredito': estadoCredito,
        'activo': activo,
      };
}

